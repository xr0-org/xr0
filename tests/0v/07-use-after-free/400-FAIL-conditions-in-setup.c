#ifdef XR0

void
modify2(int *q, int x) ~ [
	setup: if (x) { q = .clump(1); } /* ERROR: setup must be decidable */
	if (x) {
		*q = 2;
	}
];

#endif

void
modify2(int *q, int x)
{
	if (x) {
		*q = 2;
	}
}
