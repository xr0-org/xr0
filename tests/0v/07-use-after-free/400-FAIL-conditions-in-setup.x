void
modify2(int *q, int x) ~ [
	setup: if (x) { q = .clump(1); } /* ERROR: setup must be decidable */
	if (x) {
		*q = 2;
	}
] {
	if (x) {
		*q = 2;
	}
}
