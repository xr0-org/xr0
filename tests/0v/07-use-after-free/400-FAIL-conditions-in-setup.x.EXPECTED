07-use-after-free/400-FAIL-conditions-in-setup.x:6:3: setup preconditions must be decidable (modify2)
